### DISCUSSION TITLE
Mixit 2023 🦁

### DISCUSSION THREAD

**Discussion by Slashgear** [lyonjs/shortvid.io#100] [Conference / Meetup Communication]
We can generate videos / gif / images for helping Mixit communication 👋🏼
Let's keep this discussion open to list what conference organizer needs or want and we will see what we could do to help ℹ️

PR #90 has been set up to handle a design for Mixit 🫶🏼

**Comment by clacote** [comment]
Cyril for MiXiT (2023) here.

Our immediate need would be for sponsors, that we need to communicate ASAP. As far as I can tell a Sponsor animation is not yet customized per event (only LyonJS for now).

Then, general needs.

About the generated output. I have a doubt that videos are properly embedded on every social medias: we would like them to autoplay (that’s the point of having animations for social posts). Do they autoplay on Twitter, LinkedIn, Mastodon? Generating GIF would avoid this issue, I think. But I’m not sure if the missing autoplay for video is a real issue.
Ideally, we could generate the requested format online, without installing anything on local. I’m not sure it is possible at the moment.

**Reply by Slashgear** [reply]
Hey ! @clacote 

ℹ️ 
- Autoplay is normally on if video are muted or with no sound (I recommend muted videos for twitter/linkedin/mastodon/...) announcements.
- If video format does not suit, we could use a "screenshot" of one frame in JPG/PNG format.
- As we want to help you on that, we could generate it on our side with a custom design template just for you ❤️ . With @CruuzAzul we could inspire from your website colors and assets to made up something.
- We will need to be able to share the outputed videos and we will share you the link after.

You could prepare 2 things for us in order to le us generate all videos at once.
- A JSON file with a table of info for each sponsor

```typescript
export interface Sponsor {
    picture: string; // Url of accessible speaker picture (we could host them or we could use logo hosted on your website)
    name: string; // will also be use as file name
    category: string // Text that describe the group of sponsor (for example: Gold, Tickets, etc...)
}
```

- Access to all assets needed for sponsor logo (if you have SVG logo that would be perfect but we know that you probably won't have them all 😅 ) 

If you have any idea of design our assets that we could/should use, could you upload them in reply here ? 
For example Pictures taken from previous edition.

💡 We could take the idea of Snowcamp Grenoble picture, with random picture for each sponsor in a set of picture of previous editons.

https://twitter.com/Slashgear_/status/1617794334656536580/video/1

**Reply by clacote** [reply]
Sounds super good to me. Alright for video.

Gonna reply here with:
- JSON data for sponsors
- their logo (they usually are SVG)

Only pain point: we don’t have every sponsors yet (we wait for them to pay to announce), so it won’t be one shot.
But I suppose I could generate the next wave without bothering you, if the tool exist.

No need for fancy assets, the one you already have as background for « MiXiT conference » is already nice. Gonna provide a #MiXiT23 logo for sponsor, though.

**Reply by Slashgear** [reply]
@clacote 

Yes, I will link you the command to let you generate last sponsor videos after confirmation of Sponsor.

Do you prefer 1:1 ratio or 16:9 ? 😄

**Reply by clacote** [reply]
1:1 ratio is probably easier with various sponsor logos having different orientations, and also better for browsing social media on mobile.

**Reply by Slashgear** [reply]
I did this on the #90 branch.
Animation is quite simple for now, maybe @CruuzAzul will have ideas to enhance it if you want.
(maybe a smooth zoom effect on the background picture with a longer display time) 


https://user-images.githubusercontent.com/6263857/215292427-8fd2da6a-cd2f-4e87-b75d-68063d684551.mp4

It use this JSON as input

```json
{
  "picture": "https://mixitconf.org/images/sponsors/logo-viseo.svg",
  "category": "tour de cou",
  "background": "https://live.staticflickr.com/65535/52097860061_1609b2a1c6_b_d.jpg",
  "name": "viseo",
}
```

**Reply by clacote** [reply]
I like what you did!
I kind of miss the nice animation you already wrote, with with Lyon skyline (shown on https://social-video-generator-git-mixit23-lyonjs.vercel.app/conference for `Mixit` conference), that was quite nice:
<img width="401" alt="mixit-skyline-animation" src="https://user-images.githubusercontent.com/750715/215293150-c6755ca7-f927-4245-a4d6-1f3671f060ee.png">

Could we reuse it for sponsors animations? 😇

**Reply by clacote** [reply]
Here is the promised data for the first batch of MiXiT23 sponsors:
```json
[
  {
    "name": "Energy Pool",
    "picture": "https://mixitconf.org/images/sponsors/logo-energy-pool.png",
    "category": "badge et stand"
  },
  {
    "name": "VISEO",
    "picture": "https://mixitconf.org/images/sponsors/logo-viseo.svg",
    "category": "badge et stand"
  },
  {
    "name": "Talan",
    "picture": "https://mixitconf.org/images/sponsors/logo-talan.svg",
    "category": "partenaire"
  },
  {
    "name": "Red Froggy",
    "picture": "https://mixitconf.org/images/sponsors/logo-redfroggy.svg",
    "category": "partenaire"
  }
]
```

If you were to need references, here is the official list of published sponsors: https://mixitconf.org/sponsors

If you were needing to steal assets from our website, they're all stored here: https://github.com/mixitconf/mixit/tree/master/src/main/images

**Reply by Slashgear** [reply]
I suppose you did not see it but I added a props named `background` to let you pass URL of any picture for background in the intro.
For example you could pass url of image hosted on your flickr and use a different one for each video 💡 


https://user-images.githubusercontent.com/6263857/215293650-808434a0-7b3a-41ef-bce5-4f3a22caaedf.mp4

**Reply by Slashgear** [reply]
Here is the outputed videos with this template for the current list of sponsors

https://mega.nz/folder/V2QVCSSI#QEXl_eIQs74lbyC7wJ9tlw


To generate new ones:
- Fetch repository
- install dependencies (pnpm, yarn or npm)
- fetch branch related to #90 
- Run remotion command

```console
pnpm remotion render src/index.tsx Mixit-Sponsor out/example.mp4 --props='{
    "name": "VISEO",
    "picture": "https://mixitconf.org/images/sponsors/logo-viseo.svg",
    "background": "https://live.staticflickr.com/65535/52097860061_1609b2a1c6_b_d.jpg",
    "category": "badge et stand"
  }'
```

With a small script, you can easily loop over JSON array file to generate all video at once if you need.

We don't provide backend generation for now, this project is under heavy construction.

**Reply by mickaelalvs** [reply]
Hi ! 👋🏼 Yes I think we can improve a bit on the different animations, but this already gives you a first visual of what we can do ! I have some ideas, I'll have a look at it if you want 😊

**Comment by mickaelalvs** [comment]
Hello hello ! 👋🏼

J'ai fait une nouvelle version pour les vidéos d'annonce des talks, je vous laisse me dire ce que vous en pensez 😊


https://user-images.githubusercontent.com/60877626/216814438-657d9700-50a3-4cf0-81ae-da6de73771a5.mp4

**Reply by Slashgear** [reply]
@clacote ça vous irait pour annoncer les speakers ? 
On peut aussi vous fournir la dernière frame au format JPEG

**Reply by clacote** [reply]
Oui ça me semble très bien, largement au dessus de notre _production value_ habituelle 😅👏

Ceci dit, je ne peux pas vous garantir que nous allons l’utiliser : jusqu’à présent on ne tweetait pas tous les talks (une centaine sur deux jours ça serait trop), seulement quelqu’un sur lesquels on communiquait, mais ça restait une exception.
Je ne veux donc pas vous faire passer trop de temps là-dessus.

**Reply by Slashgear** [reply]
Ce que Touraine Tech a fait et qui a pas mal marché, ils ont fourni les vidéos aux speakers dans un drive pour que chacun puisse s'en servir pour annoncer/rappeler leur passage. Cela vous évite de faire la com' vous même et ça permet de faire parler de Mixit au travers des réseaux des différents speakers :) (qui n'ont souvent pas le temps de faire un tweet/post shiny pour annoncer leur passage, mais avec ça il ont la matière)

Maintenant qu'on a le template de fait, il nous suffit d'avoir le JSON avec le bon format et on génère toutes les videos en moins de 20 minutes. 😅 

Est-ce que vous avez besoin de générer des affichages pour le lieu de la conf ? Notre outil s'y prete normalement assez bien aussi vu qu'on peut générer du JPEG

**Reply by Slashgear** [reply]
@clacote Est-ce que cela vous intéresse ? :)

**Reply by clacote** [reply]
Oui c’est une bonne idée de mettre les animations à disposition des speakers eux-mêmes.
Des animations pour les affichages sur le lieu, je ne pense pas. On préfère je pense garder la main avec des slides qu’on ajuste rapidement.

**Reply by Slashgear** [reply]
@clacote Vous avez une API sur laquelle on peut récupérer la liste des talks avec les speakers pour générer ça ? :)

**Reply by clacote** [reply]
Oui:
https://mixitconf.org/en/api/2023/talk
https://mixitconf.org/en/api/2023/speaker

Attention le programme 2023 est encore en mouvement.

**Reply by Slashgear** [reply]
Yes, on va préparer des trucs mais on attendra votre Go avant de tout générer.
On vous déposera ça sur un drive/mega pour que vous puissiez le partager à tous les speakers.

**Reply by Slashgear** [reply]
Je nous ai fait un petit script JS pour générer le JSON qui nous permettra de générer toutes les vidéos d'un seul coup.
Je laisse @CruuzAzul le review.

https://github.com/lyonjs/social-video-generator/pull/145

Dès que votre programme est stabilisé/confirmé coté Mixit @clacote on lance la génération de tout ça :)

**Reply by Slashgear** [reply]
@clacote Est-ce que le programme est plus stable maintenant ou doit-on attendre encore un peu ?

**Reply by clacote** [reply]
Le programme est probablement stable pour les conférences normales, mais il manque encore toutes les keynotes.

**Reply by Slashgear** [reply]
Voilà le lien de toutes les vidéos générées pour le moment. @clacote 

https://mega.nz/folder/hvwAwa5R#gv1Wclj5WwKR8ER0BoYmaQ

Ce lien peut être partagé aux speakers afin qu'il s'en servent pour communiquer sur les réseaux leur participation.
Je supprimerai les vidéos en juin prochain.

Si il y a des soucis sur une vidéo, les speakers peuvent la regénérer eux même en se rendant sur cette page.

https://social-video-generator.vercel.app/conferences/Mixit

Tout est bon ?
